Interface.js

```diff
@@ -7,104 +7,73 @@ define('Interface', ['jQuery',
 	'Problems',
 	'jQueryTmpl',
 	'ModesConversion',
-	'Declaration'], function(){
+	'Declaration',
+	'Misc'
+], function () {
 	var Problems = require('Problems');
 	var ModesConversion = require('ModesConversion');
 	var Servers = require('Servers');
 
-	function login(callback, firstTrying){
-		currentServer.setSid(undefined);
-		currentServer.loginRequest(currentServer.user.login, currentServer.user.passwd, function(data) {
-			if (data.status == 'ok') {
-				currentServer.setSid(data.sid);
+	function login(login, passwd, callback) {
+		currentServer.loginRequest(login, passwd, function (data) {
+			if (data.status != 'ok') {
+				return showLoginForm();
 			}
-			else if(firstTrying){
-				$("#enterPassword").bind("dialogbeforeclose", function(event, ui) {
-					if (currentServer.getSid())
-						showNewUser();
-					$("#enterPassword").bind("dialogbeforeclose", function(event, ui){});
-				});
-				$('#enterPassword').dialog('open') ;
-				return;
-			}
-			else{
-				alert(data.message);
-				return false;
-			}
-			$.cookie('passwd', currentServer.user.passwd);
-			if(currentServer.user.jury){
-				currentServer.user.passwd = '';
-				$('#password').prop('value', '');
-				for (var i = 0; i < problems.length; ++i)
-					$('#forJury' + i).show();
-			}
-			//logined = true;
-			if (callback)
-				callback();
-			return true;
+			$.cookie('sid', data.sid);
+			currentServer.authenticateBySid(data.sid, function () {
+				fillTabs();
+
+				var user = currentServer.getUser();
+				user.login = login;
+				user.passwd = passwd;

```

Comment:
Don't think that's an issue, 'cos there's already was something like that in prev implementation. And it's already session-wise, not permanent.